network/manager.go

```diff
@@ -324,9 +309,33 @@ func (m *Manager) watchNetworks() {
 	}
 }
 
-func (m *Manager) Run() {
+func (m *Manager) Run(ctx context.Context) {
 	wg := sync.WaitGroup{}
 
+	if m.isMultiNetwork() {
+		for {
+			// Try adding initial networks
+			if result, err := m.sm.WatchNetworks(ctx, nil); err == nil {
```

Comment:
Looks much better. But like Jon said, it's better to log the error to help the user diagnose the problem (even though in the case of etcd not being up, the error is confusing). Logging that a retry is coming is less useful.